// @aws-sdk/client-elasticsearch-service/dist-es/commands/AcceptInboundCrossClusterSearchConnectionCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { AcceptInboundCrossClusterSearchConnectionRequest, AcceptInboundCrossClusterSearchConnectionResponse, } from "../models/models_0";
import { deserializeAws_restJson1AcceptInboundCrossClusterSearchConnectionCommand, serializeAws_restJson1AcceptInboundCrossClusterSearchConnectionCommand, } from "../protocols/Aws_restJson1";
var AcceptInboundCrossClusterSearchConnectionCommand = (function (_super) {
    __extends(AcceptInboundCrossClusterSearchConnectionCommand, _super);
    function AcceptInboundCrossClusterSearchConnectionCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    AcceptInboundCrossClusterSearchConnectionCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "ElasticsearchServiceClient";
        var commandName = "AcceptInboundCrossClusterSearchConnectionCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: AcceptInboundCrossClusterSearchConnectionRequest.filterSensitiveLog,
            outputFilterSensitiveLog: AcceptInboundCrossClusterSearchConnectionResponse.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    AcceptInboundCrossClusterSearchConnectionCommand.prototype.serialize = function (input, context) {
        return serializeAws_restJson1AcceptInboundCrossClusterSearchConnectionCommand(input, context);
    };
    AcceptInboundCrossClusterSearchConnectionCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_restJson1AcceptInboundCrossClusterSearchConnectionCommand(output, context);
    };
    return AcceptInboundCrossClusterSearchConnectionCommand;
}($Command));
export { AcceptInboundCrossClusterSearchConnectionCommand };
