// @aws-sdk/client-elasticsearch-service/dist-es/commands/CreateOutboundCrossClusterSearchConnectionCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { CreateOutboundCrossClusterSearchConnectionRequest, CreateOutboundCrossClusterSearchConnectionResponse, } from "../models/models_0";
import { deserializeAws_restJson1CreateOutboundCrossClusterSearchConnectionCommand, serializeAws_restJson1CreateOutboundCrossClusterSearchConnectionCommand, } from "../protocols/Aws_restJson1";
var CreateOutboundCrossClusterSearchConnectionCommand = (function (_super) {
    __extends(CreateOutboundCrossClusterSearchConnectionCommand, _super);
    function CreateOutboundCrossClusterSearchConnectionCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    CreateOutboundCrossClusterSearchConnectionCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "ElasticsearchServiceClient";
        var commandName = "CreateOutboundCrossClusterSearchConnectionCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: CreateOutboundCrossClusterSearchConnectionRequest.filterSensitiveLog,
            outputFilterSensitiveLog: CreateOutboundCrossClusterSearchConnectionResponse.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    CreateOutboundCrossClusterSearchConnectionCommand.prototype.serialize = function (input, context) {
        return serializeAws_restJson1CreateOutboundCrossClusterSearchConnectionCommand(input, context);
    };
    CreateOutboundCrossClusterSearchConnectionCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_restJson1CreateOutboundCrossClusterSearchConnectionCommand(output, context);
    };
    return CreateOutboundCrossClusterSearchConnectionCommand;
}($Command));
export { CreateOutboundCrossClusterSearchConnectionCommand };
