// @aws-sdk/client-elasticsearch-service/dist-es/commands/DeleteInboundCrossClusterSearchConnectionCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { DeleteInboundCrossClusterSearchConnectionRequest, DeleteInboundCrossClusterSearchConnectionResponse, } from "../models/models_0";
import { deserializeAws_restJson1DeleteInboundCrossClusterSearchConnectionCommand, serializeAws_restJson1DeleteInboundCrossClusterSearchConnectionCommand, } from "../protocols/Aws_restJson1";
var DeleteInboundCrossClusterSearchConnectionCommand = (function (_super) {
    __extends(DeleteInboundCrossClusterSearchConnectionCommand, _super);
    function DeleteInboundCrossClusterSearchConnectionCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    DeleteInboundCrossClusterSearchConnectionCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "ElasticsearchServiceClient";
        var commandName = "DeleteInboundCrossClusterSearchConnectionCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: DeleteInboundCrossClusterSearchConnectionRequest.filterSensitiveLog,
            outputFilterSensitiveLog: DeleteInboundCrossClusterSearchConnectionResponse.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    DeleteInboundCrossClusterSearchConnectionCommand.prototype.serialize = function (input, context) {
        return serializeAws_restJson1DeleteInboundCrossClusterSearchConnectionCommand(input, context);
    };
    DeleteInboundCrossClusterSearchConnectionCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_restJson1DeleteInboundCrossClusterSearchConnectionCommand(output, context);
    };
    return DeleteInboundCrossClusterSearchConnectionCommand;
}($Command));
export { DeleteInboundCrossClusterSearchConnectionCommand };
