// @aws-sdk/client-elasticsearch-service/dist-es/commands/DeleteElasticsearchDomainCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { DeleteElasticsearchDomainRequest, DeleteElasticsearchDomainResponse } from "../models/models_0";
import { deserializeAws_restJson1DeleteElasticsearchDomainCommand, serializeAws_restJson1DeleteElasticsearchDomainCommand, } from "../protocols/Aws_restJson1";
var DeleteElasticsearchDomainCommand = (function (_super) {
    __extends(DeleteElasticsearchDomainCommand, _super);
    function DeleteElasticsearchDomainCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    DeleteElasticsearchDomainCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "ElasticsearchServiceClient";
        var commandName = "DeleteElasticsearchDomainCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: DeleteElasticsearchDomainRequest.filterSensitiveLog,
            outputFilterSensitiveLog: DeleteElasticsearchDomainResponse.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    DeleteElasticsearchDomainCommand.prototype.serialize = function (input, context) {
        return serializeAws_restJson1DeleteElasticsearchDomainCommand(input, context);
    };
    DeleteElasticsearchDomainCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_restJson1DeleteElasticsearchDomainCommand(output, context);
    };
    return DeleteElasticsearchDomainCommand;
}($Command));
export { DeleteElasticsearchDomainCommand };
