// @aws-sdk/client-elasticsearch-service/dist-es/commands/DescribeOutboundCrossClusterSearchConnectionsCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { DescribeOutboundCrossClusterSearchConnectionsRequest, DescribeOutboundCrossClusterSearchConnectionsResponse, } from "../models/models_0";
import { deserializeAws_restJson1DescribeOutboundCrossClusterSearchConnectionsCommand, serializeAws_restJson1DescribeOutboundCrossClusterSearchConnectionsCommand, } from "../protocols/Aws_restJson1";
var DescribeOutboundCrossClusterSearchConnectionsCommand = (function (_super) {
    __extends(DescribeOutboundCrossClusterSearchConnectionsCommand, _super);
    function DescribeOutboundCrossClusterSearchConnectionsCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    DescribeOutboundCrossClusterSearchConnectionsCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "ElasticsearchServiceClient";
        var commandName = "DescribeOutboundCrossClusterSearchConnectionsCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: DescribeOutboundCrossClusterSearchConnectionsRequest.filterSensitiveLog,
            outputFilterSensitiveLog: DescribeOutboundCrossClusterSearchConnectionsResponse.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    DescribeOutboundCrossClusterSearchConnectionsCommand.prototype.serialize = function (input, context) {
        return serializeAws_restJson1DescribeOutboundCrossClusterSearchConnectionsCommand(input, context);
    };
    DescribeOutboundCrossClusterSearchConnectionsCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_restJson1DescribeOutboundCrossClusterSearchConnectionsCommand(output, context);
    };
    return DescribeOutboundCrossClusterSearchConnectionsCommand;
}($Command));
export { DescribeOutboundCrossClusterSearchConnectionsCommand };
