// @aws-sdk/client-elasticsearch-service/dist-es/commands/DescribeReservedElasticsearchInstancesCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { DescribeReservedElasticsearchInstancesRequest, DescribeReservedElasticsearchInstancesResponse, } from "../models/models_0";
import { deserializeAws_restJson1DescribeReservedElasticsearchInstancesCommand, serializeAws_restJson1DescribeReservedElasticsearchInstancesCommand, } from "../protocols/Aws_restJson1";
var DescribeReservedElasticsearchInstancesCommand = (function (_super) {
    __extends(DescribeReservedElasticsearchInstancesCommand, _super);
    function DescribeReservedElasticsearchInstancesCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    DescribeReservedElasticsearchInstancesCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "ElasticsearchServiceClient";
        var commandName = "DescribeReservedElasticsearchInstancesCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: DescribeReservedElasticsearchInstancesRequest.filterSensitiveLog,
            outputFilterSensitiveLog: DescribeReservedElasticsearchInstancesResponse.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    DescribeReservedElasticsearchInstancesCommand.prototype.serialize = function (input, context) {
        return serializeAws_restJson1DescribeReservedElasticsearchInstancesCommand(input, context);
    };
    DescribeReservedElasticsearchInstancesCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_restJson1DescribeReservedElasticsearchInstancesCommand(output, context);
    };
    return DescribeReservedElasticsearchInstancesCommand;
}($Command));
export { DescribeReservedElasticsearchInstancesCommand };
