// @aws-sdk/client-elasticsearch-service/dist-es/commands/DescribeElasticsearchDomainsCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { DescribeElasticsearchDomainsRequest, DescribeElasticsearchDomainsResponse } from "../models/models_0";
import { deserializeAws_restJson1DescribeElasticsearchDomainsCommand, serializeAws_restJson1DescribeElasticsearchDomainsCommand, } from "../protocols/Aws_restJson1";
var DescribeElasticsearchDomainsCommand = (function (_super) {
    __extends(DescribeElasticsearchDomainsCommand, _super);
    function DescribeElasticsearchDomainsCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    DescribeElasticsearchDomainsCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "ElasticsearchServiceClient";
        var commandName = "DescribeElasticsearchDomainsCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: DescribeElasticsearchDomainsRequest.filterSensitiveLog,
            outputFilterSensitiveLog: DescribeElasticsearchDomainsResponse.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    DescribeElasticsearchDomainsCommand.prototype.serialize = function (input, context) {
        return serializeAws_restJson1DescribeElasticsearchDomainsCommand(input, context);
    };
    DescribeElasticsearchDomainsCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_restJson1DescribeElasticsearchDomainsCommand(output, context);
    };
    return DescribeElasticsearchDomainsCommand;
}($Command));
export { DescribeElasticsearchDomainsCommand };
