// @aws-sdk/client-elasticsearch-service/dist-es/commands/DescribeReservedElasticsearchInstanceOfferingsCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { DescribeReservedElasticsearchInstanceOfferingsRequest, DescribeReservedElasticsearchInstanceOfferingsResponse, } from "../models/models_0";
import { deserializeAws_restJson1DescribeReservedElasticsearchInstanceOfferingsCommand, serializeAws_restJson1DescribeReservedElasticsearchInstanceOfferingsCommand, } from "../protocols/Aws_restJson1";
var DescribeReservedElasticsearchInstanceOfferingsCommand = (function (_super) {
    __extends(DescribeReservedElasticsearchInstanceOfferingsCommand, _super);
    function DescribeReservedElasticsearchInstanceOfferingsCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    DescribeReservedElasticsearchInstanceOfferingsCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "ElasticsearchServiceClient";
        var commandName = "DescribeReservedElasticsearchInstanceOfferingsCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: DescribeReservedElasticsearchInstanceOfferingsRequest.filterSensitiveLog,
            outputFilterSensitiveLog: DescribeReservedElasticsearchInstanceOfferingsResponse.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    DescribeReservedElasticsearchInstanceOfferingsCommand.prototype.serialize = function (input, context) {
        return serializeAws_restJson1DescribeReservedElasticsearchInstanceOfferingsCommand(input, context);
    };
    DescribeReservedElasticsearchInstanceOfferingsCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_restJson1DescribeReservedElasticsearchInstanceOfferingsCommand(output, context);
    };
    return DescribeReservedElasticsearchInstanceOfferingsCommand;
}($Command));
export { DescribeReservedElasticsearchInstanceOfferingsCommand };
