// @aws-sdk/client-elasticsearch-service/dist-es/commands/DissociatePackageCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { DissociatePackageRequest, DissociatePackageResponse } from "../models/models_0";
import { deserializeAws_restJson1DissociatePackageCommand, serializeAws_restJson1DissociatePackageCommand, } from "../protocols/Aws_restJson1";
var DissociatePackageCommand = (function (_super) {
    __extends(DissociatePackageCommand, _super);
    function DissociatePackageCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    DissociatePackageCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "ElasticsearchServiceClient";
        var commandName = "DissociatePackageCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: DissociatePackageRequest.filterSensitiveLog,
            outputFilterSensitiveLog: DissociatePackageResponse.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    DissociatePackageCommand.prototype.serialize = function (input, context) {
        return serializeAws_restJson1DissociatePackageCommand(input, context);
    };
    DissociatePackageCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_restJson1DissociatePackageCommand(output, context);
    };
    return DissociatePackageCommand;
}($Command));
export { DissociatePackageCommand };
