// @aws-sdk/client-elasticsearch-service/dist-es/commands/GetUpgradeStatusCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { GetUpgradeStatusRequest, GetUpgradeStatusResponse } from "../models/models_0";
import { deserializeAws_restJson1GetUpgradeStatusCommand, serializeAws_restJson1GetUpgradeStatusCommand, } from "../protocols/Aws_restJson1";
var GetUpgradeStatusCommand = (function (_super) {
    __extends(GetUpgradeStatusCommand, _super);
    function GetUpgradeStatusCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    GetUpgradeStatusCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "ElasticsearchServiceClient";
        var commandName = "GetUpgradeStatusCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: GetUpgradeStatusRequest.filterSensitiveLog,
            outputFilterSensitiveLog: GetUpgradeStatusResponse.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    GetUpgradeStatusCommand.prototype.serialize = function (input, context) {
        return serializeAws_restJson1GetUpgradeStatusCommand(input, context);
    };
    GetUpgradeStatusCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_restJson1GetUpgradeStatusCommand(output, context);
    };
    return GetUpgradeStatusCommand;
}($Command));
export { GetUpgradeStatusCommand };
