// @aws-sdk/client-elasticsearch-service/dist-es/commands/ListPackagesForDomainCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { ListPackagesForDomainRequest, ListPackagesForDomainResponse } from "../models/models_0";
import { deserializeAws_restJson1ListPackagesForDomainCommand, serializeAws_restJson1ListPackagesForDomainCommand, } from "../protocols/Aws_restJson1";
var ListPackagesForDomainCommand = (function (_super) {
    __extends(ListPackagesForDomainCommand, _super);
    function ListPackagesForDomainCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    ListPackagesForDomainCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "ElasticsearchServiceClient";
        var commandName = "ListPackagesForDomainCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: ListPackagesForDomainRequest.filterSensitiveLog,
            outputFilterSensitiveLog: ListPackagesForDomainResponse.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    ListPackagesForDomainCommand.prototype.serialize = function (input, context) {
        return serializeAws_restJson1ListPackagesForDomainCommand(input, context);
    };
    ListPackagesForDomainCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_restJson1ListPackagesForDomainCommand(output, context);
    };
    return ListPackagesForDomainCommand;
}($Command));
export { ListPackagesForDomainCommand };
