// @aws-sdk/client-elasticsearch-service/dist-es/commands/ListElasticsearchInstanceTypesCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { ListElasticsearchInstanceTypesRequest, ListElasticsearchInstanceTypesResponse } from "../models/models_0";
import { deserializeAws_restJson1ListElasticsearchInstanceTypesCommand, serializeAws_restJson1ListElasticsearchInstanceTypesCommand, } from "../protocols/Aws_restJson1";
var ListElasticsearchInstanceTypesCommand = (function (_super) {
    __extends(ListElasticsearchInstanceTypesCommand, _super);
    function ListElasticsearchInstanceTypesCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    ListElasticsearchInstanceTypesCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "ElasticsearchServiceClient";
        var commandName = "ListElasticsearchInstanceTypesCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: ListElasticsearchInstanceTypesRequest.filterSensitiveLog,
            outputFilterSensitiveLog: ListElasticsearchInstanceTypesResponse.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    ListElasticsearchInstanceTypesCommand.prototype.serialize = function (input, context) {
        return serializeAws_restJson1ListElasticsearchInstanceTypesCommand(input, context);
    };
    ListElasticsearchInstanceTypesCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_restJson1ListElasticsearchInstanceTypesCommand(output, context);
    };
    return ListElasticsearchInstanceTypesCommand;
}($Command));
export { ListElasticsearchInstanceTypesCommand };
