// @aws-sdk/client-elasticsearch-service/dist-es/commands/RejectInboundCrossClusterSearchConnectionCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { RejectInboundCrossClusterSearchConnectionRequest, RejectInboundCrossClusterSearchConnectionResponse, } from "../models/models_0";
import { deserializeAws_restJson1RejectInboundCrossClusterSearchConnectionCommand, serializeAws_restJson1RejectInboundCrossClusterSearchConnectionCommand, } from "../protocols/Aws_restJson1";
var RejectInboundCrossClusterSearchConnectionCommand = (function (_super) {
    __extends(RejectInboundCrossClusterSearchConnectionCommand, _super);
    function RejectInboundCrossClusterSearchConnectionCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    RejectInboundCrossClusterSearchConnectionCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "ElasticsearchServiceClient";
        var commandName = "RejectInboundCrossClusterSearchConnectionCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: RejectInboundCrossClusterSearchConnectionRequest.filterSensitiveLog,
            outputFilterSensitiveLog: RejectInboundCrossClusterSearchConnectionResponse.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    RejectInboundCrossClusterSearchConnectionCommand.prototype.serialize = function (input, context) {
        return serializeAws_restJson1RejectInboundCrossClusterSearchConnectionCommand(input, context);
    };
    RejectInboundCrossClusterSearchConnectionCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_restJson1RejectInboundCrossClusterSearchConnectionCommand(output, context);
    };
    return RejectInboundCrossClusterSearchConnectionCommand;
}($Command));
export { RejectInboundCrossClusterSearchConnectionCommand };
