// @aws-sdk/client-elasticsearch-service/dist-es/commands/UpdatePackageCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { UpdatePackageRequest, UpdatePackageResponse } from "../models/models_0";
import { deserializeAws_restJson1UpdatePackageCommand, serializeAws_restJson1UpdatePackageCommand, } from "../protocols/Aws_restJson1";
var UpdatePackageCommand = (function (_super) {
    __extends(UpdatePackageCommand, _super);
    function UpdatePackageCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    UpdatePackageCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "ElasticsearchServiceClient";
        var commandName = "UpdatePackageCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: UpdatePackageRequest.filterSensitiveLog,
            outputFilterSensitiveLog: UpdatePackageResponse.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    UpdatePackageCommand.prototype.serialize = function (input, context) {
        return serializeAws_restJson1UpdatePackageCommand(input, context);
    };
    UpdatePackageCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_restJson1UpdatePackageCommand(output, context);
    };
    return UpdatePackageCommand;
}($Command));
export { UpdatePackageCommand };
